src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -3,356 +3,119 @@
 
 using System;
 using System.IO;
-using System.Globalization;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
 using Microsoft.AspNetCore.Http.Headers;
 using Microsoft.AspNetCore.ResponseCaching.Internal;
-using Microsoft.Extensions.Primitives;
 using Microsoft.Net.Http.Headers;
 
 namespace Microsoft.AspNetCore.ResponseCaching
 {
-    internal class ResponseCachingContext
+    public class ResponseCachingContext
     {
-        private readonly HttpContext _httpContext;
-        private readonly IResponseCache _cache;
-        private readonly ResponseCachingOptions _options;
-        private readonly ICacheabilityValidator _cacheabilityValidator;
-        private readonly IKeyProvider _keyProvider;
+        private static readonly CacheControlHeaderValue EmptyCacheControl = new CacheControlHeaderValue();
 
-        private ResponseCachingState _state;
+        private RequestHeaders _requestHeaders;
+        private ResponseHeaders _responseHeaders;
+        private CacheControlHeaderValue _requestCacheControl;
+        private CacheControlHeaderValue _responseCacheControl;
 
         internal ResponseCachingContext(
             HttpContext httpContext,
             IResponseCache cache,
             ResponseCachingOptions options,
             ICacheabilityValidator cacheabilityValidator,
-            IKeyProvider keyProvider)
+            ICacheKeyProvider keyProvider)
         {
-            _httpContext = httpContext;
-            _cache = cache;
-            _options = options;
-            _cacheabilityValidator = cacheabilityValidator;
-            _keyProvider = keyProvider;
+            HttpContext = httpContext;
+            Cache = cache;
+            Options = options;
+            CacheabilityValidator = cacheabilityValidator;
+            KeyProvider = keyProvider;
         }
 
-        internal ResponseCachingState State
-        {
-            get
-            {
-                if (_state == null)
-                {
-                    _state = _httpContext.GetResponseCachingState();
-                }
-                return _state;
-            }
-        }
-
-        internal bool ResponseStarted { get; set; }
-
-        private Stream OriginalResponseStream { get; set; }
-
-        private ResponseCacheStream ResponseCacheStream { get; set; }
+        public HttpContext HttpContext { get; }
 
-        private IHttpSendFileFeature OriginalSendFileFeature { get; set; }
+        internal IResponseCache Cache { get; }
```

Comment:
Yea, but is there any reason to expose them here? Just resolve it from DI if you need it.